net-dns/knot: Add quic USE flag, conflict with net-libs/ngtcp2

nemunaire opened the pull request:
This PR address the bug https://bugs.gentoo.org/878297

Here are the changes:

```diff
--- knot-3.2.1-r1.ebuild
+++ knot-3.2.1-r2.ebuild
@@ -24,6 +24,7 @@
 	"+noudp"
 	"+onlinesign"
 	"+queryacl"
+	"quic"
 	"+rrl"
 	"+stats"
 	"+synthrecord"
@@ -50,6 +51,10 @@
 		!libidn2? ( net-dns/libidn:0= !net-dns/libidn2 )
 		libidn2? ( net-dns/libidn2:= )
 	)
+	quic? (
+		  net-libs/gnutls:=
+		  !net-libs/ngtcp2
+	)
 	systemd? ( sys-apps/systemd:= )
 	xdp? (
 		 dev-libs/libbpf:=
@@ -72,6 +77,7 @@
 		$(use_enable fastparser)
 		$(use_enable dnstap)
 		$(use_enable doc documentation)
+		$(use_enable quic)
 		$(use_enable utils utilities)
 		$(use_enable xdp)
 		--enable-systemd=$(usex systemd)
```

thesamesam commented:
Why have you added a blocker again? I think we should try to unbundle ngctp or have it not touch the system copy at all.

nemunaire commented:
> Why have you added a blocker again? I think we should try to unbundle ngctp or have it not touch the system copy at all.

knot version 3.2.1 requires ngtcp2 in version =0.0.8 (it fails to compile with a newer version like 0.0.10) which is not available in the tree. Next version of knot is compatible with >=ngtcp2-0.0.10, it'll be possible to use the version in the tree in the next release.

For those who don't have ngtcp2 on their system, the bundled ngtcp2 version is used (I added a dependency on net-libs/gnutls, as I see it's required for QUIC support in knot).

If it's available on the system, I don't have seen a way to force the use of the bundled one: their is a possibility to force the bundled ngtcp2 through `--enable-quic=embedded`, but it can only be defined by the configure itself, in case it is not found on the system.

If their is a patch to made, that can be replacing `libngtcp2 >= 0.8.0` to `libngtcp2 == 0.8.0` in order to use the embedded lib if the system has a greater version.
The next release of knot should come in a few weeks, it seems quite risky and a waste of time to hack this version of knot in order to be compatible with a corner case, addressed and tested upstream in their next release.
